File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        int ml= 1;
        int c=1;
        int n= nums.size();
        if (n==0) return 0;
        sort(nums.begin() , nums.end());
        for (int i=1 ; i<n; i++ ){
            if (nums[i]== nums[i-1]){
                continue;
            }
            if(nums[i]-1 == nums[i-1]){
                c++;
            }
            else{
                ml= max(ml, c);
                c=1;
                
            }
        }
        return max(ml, c);;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        nums_set= set(nums)
        count =0
        max_count= -1

        for i in nums_set:
            if (i-1 in nums_set):
                pass

            else:
                next_element= i+1
                count= 0
                while (next_element in nums_set):
                    count+=1
                    next_element+=1

                max_count= max(max_count, count)

        return max_count+1

                

        